weapons/environment-weapons.yaml

```diff
@@ -68,47 +68,68 @@ Tiberium:
 		Falloff: 100, 0
 		Spread: 7c73
 		DamageTypes: Repair
+		ValidStances: Ally
 	Warhead@Condition: GrantExternalCondition
 		Condition: FeatureBuffRepair
 		Spread: 7c73
 		Duration: 1
+		ValidStances: Ally
 
 ^GlobalRepair:
-	Warhead@Damage: SpreadDamage
-		Falloff: 100, 0
-		Spread: 1000000c0
+	ReloadDelay: 1
+	Warhead@Global: GlobalExplodeWeapon
+		ValidStances: Ally
+		Weapon: ^GlobalRepair1
+
+^GlobalRepair1:
+	Warhead@Damage: TargetDamage
+		ValidStances: Ally
 	Warhead@Condition: GrantExternalCondition
 		Condition: FeatureBuffRepair
-		Spread: 1000000c0
+		Spread: 0c0
+		Duration: 1
+		ValidStances: Ally
 
 RepairInfantry:
 	ReloadDelay: 10
 	Inherits: ^Repair
-	Warhead@Damage: SpreadDamage
+	Warhead@Damage:
```

Comment:
Ah, yes, if the inherited template already defines the types of these warheads, it's all fine. I guess this will be just hader to red then.